addons/purchase/views/purchase_views.xml

```diff
@@ -621,6 +621,21 @@
             </field>
         </record>
 
+        <record id="view_product_normal_purchase_buttons_from" model="ir.ui.view">
+             <field name="name">product.product.purchase.button.inherit</field>
+             <field name="model">product.product</field>
+             <field name="inherit_id" ref="product.product_normal_form_view"/>
+             <field name="groups_id" eval="[(4, ref('purchase.group_purchase_user'))]"/>
+             <field name="arch" type="xml">
+                 <div name="button_box" position="inside">
+                    <button class="oe_stat_button" name="%(purchase.action_purchase_line_product_tree)d"
```

Comment:
This will not work. The action `action_purchase_line_product_tree` expects ids of `product.template` records, not `product.product`

Reply:
You're right, I tested with only one variant so I didn't see the problem.
Well, I would create a new one action similar to `action_purchase_line_product_tree` which differs only for the `domain`.
Moreover, I'd like to know which name should be suited for this latter.

Do you agree with this proposal? Please let me know.

Reply:
You can take example of what is done in sale ([views](https://github.com/odoo/odoo/blob/f4d08ade92031f3bda2fb7648add83d3a0822a3e/addons/sale/views/sale_views.xml#L673), [code](https://github.com/odoo/odoo/blob/f4d08ade92031f3bda2fb7648add83d3a0822a3e/addons/sale/models/product_template.py#L30)).
But this should be done in master, not for stable version.

Reply:
> You can take example of what is done in sale (views, code).

Thanks a lot, @mart-e!

> But this should be done in master, not for stable version.

Uh, what a pity, these changes seem to me not so invasive  :disappointed_relieved: Could you please highlight which point of https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean shouldn't be respected?

Thank you in advance.

Reply:
The "No improvement" part. This is not strictly a bug but a view improvement. You are changing the UI to add buttons+python code which may mess with other customizations. This kind of changes will be done if we really have a problem, especially in older versions like 10.0.

Reply:
> The "No improvement" part. This is not strictly a bug but a view improvement.

One of our customer is migrating from `8.0` to `10.0` and asking us to have this feature already present in `8.0`. I just checked it in his istance `8.0` and there it works. So this seem to me "no improvement".

You'd rather open a support ticket?

Reply:
Yes but we did a big refactoring of the purchase module in 9.0 (cb01be235e73cf948418e41642d4906ff84906a6) where we simplified the views and removed some parts.
You can always try with a support ticket if they have a justification for the removal/readding.

Reply:
Thanks @mart-e, I will open a support ticket then.